dataland-frontend/tests/e2e/specs/data-ownership/DataOwnershipUserJourney.ts

```diff
@@ -0,0 +1,85 @@
+import { describeIf } from "@e2e/support/TestUtility";
+import { admin_name, admin_pw, reader_name, reader_pw } from "@e2e/utils/Cypress";
+import { getKeycloakToken, login, logout } from "@e2e/utils/Auth";
+import {
+  CompanyDataControllerApi,
+  type CompanyDataOwners,
+  Configuration,
+  DataTypeEnum,
+  type PathwaysToParisData,
+} from "@clients/backend";
+import { generateDummyCompanyInformation, uploadCompanyViaApi } from "@e2e/utils/CompanyUpload";
+import { uploadFrameworkData } from "@e2e/utils/FrameworkUpload";
+import { type FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+import { ARRAY_OF_FRAMEWORKS_WITH_UPLOAD_FORM } from "@/utils/Constants";
+
+let p2pFixtureForTest: FixtureData<PathwaysToParisData>;
+before(function () {
+  cy.fixture("CompanyInformationWithP2pPreparedFixtures").then(function (jsonContent) {
+    const preparedFixturesP2p = jsonContent as Array<FixtureData<PathwaysToParisData>>;
+    p2pFixtureForTest = getPreparedFixture("P2p-dataset-with-no-null-fields", preparedFixturesP2p);
+  });
+});
+
+describeIf(
+  "As a user, I expect to be able to upload data for one company for which iam data owner",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+  },
+  function (): void {
+    beforeEach(() => {
+      cy.ensureLoggedIn(admin_name, admin_pw);
+    });
+
+    it("Upload a company, set a user as the data owner and then verify that the upload pages are displayed for that user", () => {
+      const uniqueCompanyMarker = Date.now().toString();
+      const testCompanyName = "Company-Created-In-Data-Owner-Test-" + uniqueCompanyMarker;
+      getKeycloakToken(admin_name, admin_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompanyName)).then((storedCompany) => {
+          return uploadFrameworkData(
+            DataTypeEnum.P2p,
+            token,
+            storedCompany.companyId,
+            "2021",
+            p2pFixtureForTest.t,
+          ).then(() => {
+            void postDataOwner(token, "18b67ecc-1176-4506-8414-1e81661017ca", storedCompany.companyId);
+            logout();
```

Comment:
I've also tried to just use some of the code from login() and logout() to reduce duplication, but the test was not faster with the reduced code. Therefore i have left in the login() and logout() methods.

Reply:
I don't understand

Reply:
I can explain it to you in a call